content/api/commands/origin.md

```diff
@@ -0,0 +1,512 @@
+to `true` in the Cypress config.
+
+Enabling this flag does the following:
+
+- It adds the [`cy.session()`](/api/commands/session) and `cy.origin()`
+  commands, and [`Cypress.session`](/api/cypress-api/session) API.
+- It adds the following new behaviors (that will be the default in a future
+  major update of Cypress) at the beginning of each test:
+  - The page is cleared (by setting it to `about:blank`).
+  - All active session data (cookies, `localStorage` and `sessionStorage`)
+    across all domains are cleared.
+- It overrides
+  the [`Cypress.Cookies.preserveOnce()`](/api/cypress-api/cookies#Preserve-Once) and
+  [`Cypress.Cookies.defaults()`](/api/cypress-api/cookies#Defaults) methods.
+- Cross-domain requests will no longer fail immediately, but instead, time out
+  based on [`pageLoadTimeout`](/guides/references/configuration#Timeouts).
+- Tests will no longer wait on page loads before moving on to the next test.
+
+Because the page is cleared before each
+test, [`cy.visit()`](/api/commands/visit) must be explicitly called in each test
+to visit a page in your application.
+
+</Alert>
+
+## Syntax
+
+```js
+cy.origin(url, callbackFn)
+cy.origin(url, options, callbackFn)
+```
+
+### Usage
+
+**<Icon name="check-circle" color="green"></Icon> Correct Usage**
+
+```js
+const hits = getHits() // Defined elsewhere
+// Run commands in secondary origin, passing in serializable values
+cy.origin('https://www.acme.com', { args: { hits } }, ({ hits }) => {
+  // Inside callback baseUrl is https://www.acme.com
+  cy.visit('/history/founder')
+  // Commands are executed in secondary origin
+  cy.get('h1').contains('About our Founder, Marvin Acme')
+  // Passed in values are accessed via callback args
+  cy.get('#hitcounter').contains(hits)
+})
+// Even though we're outside the secondary origin block,
+// we're still on acme.com so return to baseUrl
+cy.visit('/')
+// Continue running commands on primary origin
+cy.get('h1').contains('My cool site under test')
+```
+
+**<Icon name="exclamation-triangle" color="red"></Icon> Incorrect Usage**
+
+```js
+const hits = getHits()
+// cy.visit() should be inside cy.origin() callback
+cy.visit('https://www.acme.com/history/founder')
+cy.origin('https://www.acme.com', () => {
+  // Fails because origin was visited before cy.origin() block
+  cy.get('h1').contains('About our Founder, Marvin Acme')
+  // Fails because hits is not passed in via args
+  cy.get('#hitcounter').contains(hits)
+})
+// Won't work because still on acme.com
+cy.get('h1').contains('My cool site under test')
+```
+
+### Arguments
+
+**<Icon name="angle-right"></Icon> url** **_(String)_**
+
+A URL specifying the secondary origin in which the callback is to be executed.
+This should at the very least contain a hostname, and may also include the
+protocol, port number & path. Query params are not supported.
+
+This argument will be used in two ways:
+
+1. It uniquely identifies a secondary origin in which the commands in the
```

Comment:
should we be using secondary origin or different origin throughout these docs? technically they could visit a t3rd or 4th and this terminology sounds like only a 2nd is supported

Reply:
I agree this could be confusing. We do want to make clear the binary distinction between the primary origin and secondary origins without folks confusing them with being ordinal. I'm not sure if primary/different is a good pairing. Maybe primary/non-primary? 

I feel like we should define these terms somewhere early in this doc, so users have that context as we use them throughout.